### PR TITLE
feat: add support for Llama 3.1

### PR DESCRIPTION
**Pull request by quitrk**

### CODE REVIEW

File: skynet/env.py [hunk 1]
```diff
@@ -26,16 +26,16 @@ def tobool(val: str | None):
 
 # models
 llama_path = os.environ.get('LLAMA_PATH')
-llama_n_ctx = int(os.environ.get('LLAMA_N_CTX', 8192))
-llama_n_gpu_layers = int(os.environ.get('LLAMA_N_GPU_LAYERS', -1 if is_mac else 40))
+llama_n_ctx = int(os.environ.get('LLAMA_N_CTX', 32000))
```
Comment (saghul) [inline skynet/env.py:29]: Is this as large as we can go on the A10? Just a thought: the default should be what the model can do, and we can tweak it as we deploy it.

File: skynet/modules/ttt/openai_api/app.py [hunk 2]
```diff
@@ -23,10 +23,11 @@ def initialize():
 
     proc = subprocess.Popen(
         f'{openai_api_server_path} \
-            -m {llama_path} \
-            -b {llama_n_batch} \
-            -c {llama_n_ctx} \
-            -ngl {llama_n_gpu_layers} \
+            --batch-size {llama_n_batch} \
+            --ctx-size {llama_n_ctx} \
+            --flash-attn \
```
Comment (saghul) [inline skynet/modules/ttt/openai_api/app.py:28]: What effect did you observe here?

File: skynet/modules/ttt/summaries/processor.py [hunk 3]
```diff
@@ -71,8 +71,12 @@ async def process(payload: DocumentPayload, job_type: JobType, model: ChatOpenAI
         chain = load_summarize_chain(current_model, chain_type="map_reduce", combine_prompt=prompt, map_prompt=prompt)
 
     result = await chain.ainvoke(input={"input_documents": docs})
+    formatted_result = result['output_text'].replace('Response:', '', 1).strip()
```
Comment (saghul) [inline skynet/modules/ttt/summaries/processor.py:74]: Why is this necessary now?

File: skynet/modules/ttt/summaries/prompts/summary.py [hunk 4]
```diff
@@ -1,27 +1,11 @@
 summary_conversation_prompt = """
-    ## Instructions
-    Provide a summary of the given conversation, following the instructions
-    The summary should be just plain text.
-    The only formatting allowed is adding a new line between main ideas.
-    Do not add any other formatting, such as bullet points, numbering, or asterisks.
-    Start your response with "Summary:".
-
-    ## Transcript
+    Summarize the following conversation. Start your response with "Response:".
```
Comment (saghul) [inline skynet/modules/ttt/summaries/prompts/summary.py:2]: Ah I see now.

**Review by saghul — APPROVED** [review]
LGTM with a coupple of minor comments.

File: skynet/modules/ttt/openai_api/app.py (hunk 2 above)
Comment (quitrk) [reply skynet/modules/ttt/openai_api/app.py:28]: https://www.hopsworks.ai/dictionary/flash-attention

As a result the available space for kv cache increases and there's some improvement with regards to the inference speed

File: skynet/modules/ttt/summaries/processor.py (hunk 3 above)
Comment (quitrk) [reply skynet/modules/ttt/summaries/processor.py:74]: I added this in order to minimise randomness in generated responses (e.g.: `sure, here's a summary of...`)

**Review by saghul — APPROVED** [review]
